src/components/DatePicker/index.jsx

```diff
@@ -1,9 +1,232 @@
 import React, { Component } from 'react';
+import moment from 'moment';
+import InfiniteCalendar, {
+    Calendar,
+    withRange,
+} from 'react-infinite-calendar';
+import 'react-infinite-calendar/styles.css';
+import {EVENT_TYPE} from "react-infinite-calendar/es/Calendar/withRange";
+import Timekeeper from "react-timekeeper";
 
+const CalendarWithRange = withRange(Calendar);
+
+function getTheme() {
+    return {
+        accentColor: '#009384',
+        floatingNav: {
+            background: '#5B57A6',
+            chevron: '#FFF',
+            color: '#FFF',
+        },
+        headerColor: '#5B57A6',
+        selectionColor: '#009384',
+        textColor: {
+            active: '#FFF',
+            default: '#231F20',
+        },
+        todayColor: '#EEA616',
+        weekdayColor: '#5B57A6',
+    };
+}
+
+function getTimePickerTheme() {
+    return {
+        // main container
+        TIMEPICKER_BACKGROUND: '#F2F2F2',
+        FONT_FAMILY: '"Roboto", sans-serif',
+        DONE_BUTTON_COLOR: '#686868',
+        DONE_BUTTON_BORDER_COLOR: '#CCC',
+
+        // time
+        TIME_BACKGROUND: '#5B57A6',
+        TIME_DEFAULT_COLOR: '#CCC',
+        TIME_SELECTED_COLOR: 'white',
+
+        // time dropdown
+        DROPDOWN_BORDER: '#f4f4f4',
+        DROPDOWN_COLOR: '#8c8c8c',
+        DROPDOWN_SELECTED_COLOR: '#5B57A6',
+
+        // clock wrapper
+        CLOCK_WRAPPER_BACKGROUND: '#f4f4f4',
+        CLOCK_WRAPPER_MERIDIEM_BACKGROUND: '#8c8c8c',
+        CLOCK_WRAPPER_MERIDIEM_COLOR: 'white',
+        CLOCK_WRAPPER_MERIDIEM_COLOR_SELECTED: '#009384',
+
+        // clock
+        CLOCK_BACKGROUND: '#FFF',
+        CLOCK_NUMBER_COLOR: '#BBB',
+        CLOCK_HAND_ARM: '#009384',
+        CLOCK_HAND_CIRCLE_BACKGROUND: '#009384',
+        CLOCK_HAND_INTERMEDIATE_CIRCLE_BACKGROUND: '#fff'
+    }
+}
+
+//props:
+//pickerType: Either "simple" or "ranged" default: simple
+//width: the DatePicker width default: 400
+//
+//hasTime: boolean default: false
+//_onSelect(date): function The date object is a moment object, or in the case of a ranged DatePicker, an array of two moment objects
+//disabledDates: An array of Dates to disable.
 class DatePicker extends Component {
+    constructor() {
+        super();
+        this.state = {
+            timeVisible: false,
+            selected: moment().startOf("day")
+        };
+        this.timePicked = null;
+        this.rangedTimeCurrentlyBeingSelected = 0;
+    }
+
     render(){
-        return(<div>Datepicker - WIP</div>);
+        switch(this.props.pickerType) {
+            case "simple":
```

Comment:
Refactor the rendering logic into separate functions and call them for each form of the component. It would be better to have a Boolean prop isRanged instead of a picker type which is text.